### ISSUE TITLE
Table Entities (only upmost row) not shown in expected order.

### ISSUE THREAD

**Issue by RoSchmi** [Microsoft/AzureStorageExplorer#1135]
Storage Explorer Version 1.6.2
Platform: Windows 10
When opening a table in Storage explorer I expect the rows sorted in the order of the contents of PartitionKey and RowKey.
Unexpectedly the upmost row does not follow this pattern. For example in a table in which all Entities have the same PartitionKey, the Entity with the lowest content should appear upmost. This is not the case. It can easily be handled by sorting the rows according to their RowKey but it is unexpected and in some way irritating.

**Comment by craxal** [comment]
Can you tell us what the contents of the PartitionKey and RowKey properties are of the first few entities, including the "upmost" row? As far as I can tell, if you click one of the headers to sort, we're only doing a simple JavaScript string comparison. It's likely we're listing entities in the order _they were received_ by default, not by partition or row key.

**Comment by RoSchmi** [comment]
Thanks for your interest,
as you can see on the attached picture not the last received row is in the upmost position. The upmost row is from the time 15.21:36, the next row is the one from the latest time 22:21:36. My RowKey are created in a way that they are in the order of ascending time. When I click on the header of the RowKey the rows get sorted as expected.
Btw: When you use the old Azure Storage Explorer 6, the rows are by default sorted as expected.
This is not really a big disadvantage but at least in some way unusual.
![tableview](https://user-images.githubusercontent.com/24595510/52745906-9033bb80-2fe0-11e9-8c06-57653b16623e.png)

**Comment by craxal** [comment]
Yeah, that is a little strange. Again, my guess, even after sorting by PartitionKey, the suborder is based on the order in which the entities were received from Azure.

**Comment by RoSchmi** [comment]
Hi, thanks for your interest.
The order in which the entities are received from Azure propably depends on the client method which is used to retrieve them from Azure.
I have made two iOS Xamarin.Forms Apps ( AzureTabStorClient and Charts4Azure ) where I use the Windows.Azure.Storage Library and the PartitionRangeQueryAsync method.
This command at least leads to receiving the entities in the expected order (see attached picture).
About your guess that after sorting by PartitionKey the suborder is based on the order in which the entities were received from Azure I can only say that the suborder at least is not in the order they were sent to Azure. If you want I can give you the key of my Account that you can see the orignal data yourself.
![tabstorclienttable](https://user-images.githubusercontent.com/24595510/52852455-f02d7d80-3118-11e9-9ee1-0bf55a13a76a.png)

**Comment by craxal** [comment]
Don't share your keys with just anyone! As trustworthy as we'd like you to believe we are, that's very unsecure, because it gives unbridled access to your account.

We use the [azure-storage-node](https://github.com/Azure/azure-storage-node) library for Storage Explorer, which works a little differently than the .NET libraries. .NET in particular probably does some preprocessing and deserialization into the proper types, which may affect the order of items. It's also possible that the .NET library submits a small query string that influences the order.

Regardless, I don't think Azure guarantees that entities will be returned in any particular order. Unless this is a serious problem for you, I don't think we're going to be able to look into this much more.

**Comment by chrome050** [comment]
The error should be corrected, because the expected behavior was not implemented or does not occur. As in my case, this can lead to several hours of troubleshooting.
